src/GraphBLAS-sharp.Backend/Algorithms/BFS.fs

```diff
@@ -0,0 +1,70 @@
+namespace GraphBLAS.FSharp.Backend.Algorithms
+
+open GraphBLAS.FSharp.Backend
+open Brahma.FSharp
+open FSharp.Quotations
+open GraphBLAS.FSharp.Backend.ArraysExtensions
+open GraphBLAS.FSharp.Backend.Common
+
+module BFS =
+    let singleSource
+        (clContext: ClContext)
+        (add: Expr<int option -> int option -> int option>)
+        (mul: Expr<'a option -> 'b option -> int option>)
+        workGroupSize
+        =
+
+        let spMVTo =
+            SpMV.runTo clContext add mul workGroupSize
+
+        let zeroCreate =
+            ClArray.zeroCreate clContext workGroupSize
+
+        let ofList = Vector.ofList clContext Dense
```

Comment:
Looks like frontier will be allocated with default flags (https://github.com/YaccConstructor/GraphBLAS-sharp/blob/net5/src/GraphBLAS-sharp.Backend/Vector/Vector.fs#L48), but actually, once initialized, it will be used on GPU only.